lib/JSON/Unmarshal.rakumod

```diff
@@ -248,7 +248,11 @@ multi _unmarshal(Any:D $json, Bool) {
    return Bool($json);
 }
 
-multi _unmarshal(%json, Mu $obj is raw) {
+multi _unmarshal(%json, @x) {
```

Comment:
There going to be a problem about this for classes trying to implement the Positional role but remain serializable otherwise (a pseudo-code example):

```raku
class MyPos does Positional {
    has @!values;
    has Str:D $.source; # Where we got our values from
    has UserData $.something-else;
    method AT-POS($idx) { self!is-ok-position ?? @!values[$idx] !! self!throw-bad-pos }
    ...
}
```

Reply:
Yes. But unfortunately without this "guard"  it will blunder into another candidate where it will blow up in some uncaught fashion.

I think another test is called for :smile_cat:

Reply:
It appears that it would have blown up in the `unmarshal(Str:D $json, Positional $obj, *%c) `  even without that candidate with something like:

```
    my class TestClass does Positional {
        has Str $.string;
    }

    my $string-attribute = '{ "string" : "test value" }';

    my $obj;

    $obj = unmarshal($string-attribute, TestClass);

    is $obj.string, 'test value', "got the attribute back";
```

On a typed attribute of another class it does fail in that candidate.  I think both can be solved with a subset which is like the reverse of the where clause you have.  Let's see.

Reply:
See the latest commit which deals with both of those cases.